Include/internal/pycore_ceval.h

```diff
@@ -154,6 +154,7 @@ extern PyObject* _Py_MakeCoro(PyFunctionObject *func);
 
 extern int _Py_HandlePending(PyThreadState *tstate);
 
+PyAPI_FUNC(PyObject *) _PyEval_GetLocals(void);
```

Comment:
Does this still need to be a PyAPI_FUNC? Since it's only used in the core, seems like we could leave it as simply `extern` like the function immediately above.

Reply:
I wasn't sure what was preferred here; there are also lots of examples of functions declared in `pycore_*.h` files (including this header) that use `PyAPI_FUNC`. But the definition of `PyAPI_FUNC` says it is for "public API functions," and I think you're right that it's not needed, so I'll remove it.

Reply:
https://devguide.python.org/developer-workflow/c-api/#with-the-extern-keyword says to use extern when in doubt

Reply:
I don't think `extern` does anything here, but some people prefer it for style reasons, and it seems like most (not all) non-static function declarations in this file use it, so I'll use it.

Reply:
If the function is to be used by extension modules that are part of the standard library (or say they are, by defining Py_BUILD_CORE), which can be built as .so files, you need to use PyAPI_FUNC. Otherwise the symbol may not be exported on Windows (and I believe a few other platforms, like Android) or when using `-fvisibility=hidden`.

FWIW, I hadn't realised you're adding a `_PyEval_GetLocals()` when there's already a `PyEval_GetLocals()` that does a subtly different thing. I'm not fond of using the same name with just a `_` difference here.

Reply:
I could make it `_PyEval_GetLocalsRef`, depending on the outcome of https://github.com/python/steering-council/issues/201, since the one notable difference is that it returns a new reference.

The other notable difference is the behavior inside an inlined comprehension in module or class scope, but I don't think there is any way to summarize that difference in a name.

This function will go away in 3.13 if we do PEP 667, or something like it, to standardize the behavior of `locals()`.

Reply:
Currently there's no use of the function in any standard library extension modules (`PyEval_GetLocals` was used only once in `object.c` and several times in `bltinmodule.c`), and I don't expect it to be needed in any, so I don't think we need to use `PyAPI_FUNC`.

Reply:
Or, I could go back to using `_PyEval_GetFrameLocals` like I did originally in this PR (but this time with a leading underscore). If we get PEP 667, it would introduce `PyEval_GetFrameLocals` as public API, and the private `_PyEval_GetFrameLocals` wouldn't be needed anymore and could be removed.